### PR TITLE
Bug/create user

### PR DESCRIPTION
**Pull request by hcnimi**
Realized passing userData object to checkUserExists is actually not a bug. Realized after creating branch, that I was .accessing email in the query. Minor styles/syntax changes.

### CODE REVIEW

File: server/index.js [hunk 1]
```diff
@@ -94,20 +94,21 @@ app.post('/signin', (req, res) => {
 
 app.post('/signup', (req, res) => {
   let userData = req.body;
+  console.log('email', userData.email);
```
Comment (skoshkarli) [inline server/index.js]: console.log!

File: server/models/index.js [hunk 2]
```diff
@@ -17,20 +17,20 @@ const createUser = (user, cb) => {
   });
 };
 
-const checkUserExists = (user, cb) => {
+const checkUserExists = (email, cb) => {
 
   let query = 'SELECT email FROM users WHERE email = ?';
-  db.connection.query(query, user.email, function(error, results, fields) {
+  db.connection.query(query, email, function(error, results, fields) {
```
Comment (skoshkarli) [inline server/models/index.js]: see the usage of this in server.js  on line 148:
db.checkUserExists(userData, (err, results) => {
    if (err) {
      res.redirect(500, '/signup');
    }
 a userdata object is being passed in to this function. this change will break fetching of the email from that object?

Comment (hcnimi) [reply server/models/index.js]: Were you looking at the "before" code :)? the "after" is db.checkUserExists(userData.email, (err, results) => {.

Comment (skoshkarli) [reply server/models/index.js]: ah yeah. sorry!

Comment (jbgomez) [inline server/models/index.js]: `email` should be inside of an array

File: server/models/index.js [hunk 3]
```diff
@@ -17,20 +17,20 @@ const createUser = (user, cb) => {
...
   });
 
 };
 
-const fetchUser = (useremail, cb) => {
+const fetchUser = (email, cb) => {
   let query = 'SELECT * FROM users WHERE email = ?';
 
-  db.connection.query(query, [useremail], (err, result, fields) => {
+  db.connection.query(query, email, (err, result, fields) => {
```
Comment (jbgomez) [inline server/models/index.js]: `email` should be inside of an array

**Comment by jbgomez** [comment]
I think we should be querying the database for users by id. It's a faster lookup and we already have that data in the session. We should update our routes, auth functions, and db models to reference the user id, as opposed to user email.

File: server/models/index.js (hunk 3 above)
Comment (hcnimi) [reply server/models/index.js]: done.

File: server/index.js [hunk 4]
```diff
@@ -140,7 +138,7 @@ app.post('/upload', upload.single('file'), function(req, res, next) {
   res.send('Successfully uploaded ' + req.file.length + ' files!')
 });
 
-app.get('*', (req, res) => {
+app.get('/*', (req, res) => {
```
Comment (skoshkarli) [inline server/index.js]: should be a star *

**Pull request merged**